internal/pusher/v2/condition.go

```diff
@@ -0,0 +1,34 @@
+package v2
+
+import (
+	"sync"
+)
+
+// condition is a simple, channel-based condition variable.
+type condition struct {
+	mu sync.Mutex
+	ch chan struct{}
+}
```

Comment:
If I'm understanding this correctly, the mutex exists so that the channel can be initialized.

While I understand the appeal of this:

```
    var ready condition
```

which is "ready to use", it does pay a price, which is that every single access has to acquire the mutex.

Instead, you could do this:

```
type condition chan struct{}

func newCondition() condition {
	return make(condition, 1)
}

func (c condition) Signal() {
	select {
	case c <- struct{}{}:
	default:
	}
}

func (c condition) C() <-chan struct{} {
	return c
}
```

this does require to change the usage to this:

```
    var cond = newCondition()
```

or

```
type foo struct {
    cond condition
    ...
}

func newFoo() foo {
   return foo{
       cond: newCondition(),
   }
}

// ...

func main() {
    myFoo := newFoo()
}
```